src/lib.rs

```diff
@@ -305,119 +291,82 @@ pub trait Compare<Lhs: ?Sized, Rhs: ?Sized = Lhs> {
     /// let f1 = &Foo { key1: 'a', key2: 2};
     /// let f2 = &Foo { key1: 'a', key2: 3};
     ///
-    /// let cmp = |lhs: &Foo, rhs: &Foo| lhs.key1.cmp(&rhs.key1);
+    /// let cmp = Extract::new(|foo: &Foo| foo.key1);
     /// assert_eq!(cmp.compare(f1, f2), Equal);
     ///
-    /// let cmp = cmp.then(|lhs: &Foo, rhs: &Foo| lhs.key2.cmp(&rhs.key2));
+    /// let cmp = cmp.then(Extract::new(|foo: &Foo| foo.key2));
```

Comment:
I agree that things are quite verbose right now, but I have some ideas for a redesign once the `unboxed_closures` feature gate goes away, and once we get custom preludes.